sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/localstorage/LocalStorageTelemetryPipelineListener.java

```diff
@@ -71,21 +74,83 @@ private void processStatusCode206(TelemetryPipelineRequest request, TelemetryPip
         Set<ResponseError> errors = response.getErrors();
         errors.forEach(error -> logger.verbose("Error in telemetry: {}", error));
         if (!errors.isEmpty()) {
-            List<TelemetryItem> originalTelemetryItems = new ArrayList<>();
-            for (ByteBuffer byteBuffer : request.getByteBuffers()) {
-                originalTelemetryItems.addAll(deserialize(byteBuffer.array()));
-            }
-            List<TelemetryItem> toBePersisted = new ArrayList<>();
+            List<ByteBuffer> originalByteBuffers = request.getByteBuffers();
+            byte[] encodedBytes = convertByteBufferListToByteArray(originalByteBuffers);
+            byte[] decodedBytes = ungzip(encodedBytes); // decode is needed in order to split by newline correctly
+            List<byte[]> decodedByteArrayList = splitBytesByNewline(decodedBytes);
+            List<ByteBuffer> toBePersisted = new ArrayList<>();
```

Comment:
no need to convert byte[] -> ByteBuffer (since we already have allocated byte[] here, no savings by wrapping it in ByteBuffer)
```suggestion
            List<byte[]> toBePersisted = new ArrayList<>();
```

Reply:
it was preparing for the gzip() which expects a list of bytebuffer